### PR TITLE
Admin tag pages

### PR DESCRIPTION
**Pull request by Allcharles**
# Admin Tag Pages

Created admin tag pages (list, new, edit, delete).

## Changes

- Created `TagsService`
- Created `Tag` model
- Created `FormlyCheckboxInput`
- Created `FormlyHorizontalWrapper`
- Created tag list/forms

## Issues

- Update forms to provide `typeOfTags` dropdown
  - Create custom resolver and service function
- Missing unit tests

## Visual Changes

Tag list page
image

New Tag Form
image

Edit Tag Form
image

Delete Tag Form
image

### CODE REVIEW

File: src/app/models/Tag.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      text: this.text,
+      count: this.count,
+      isTaxanomic: this.isTaxanomic,
+      typeOfTag: this.typeOfTag,
+      retired: this.retired,
+      notes: this.notes,
+      creatorId: this.creatorId,
+      updaterId: this.updaterId,
+      createdAt: this.createdAt?.toISO(),
+      updatedAt: this.updatedAt?.toISO()
```
Comment (atruskie) [inline src/app/models/Tag.ts:70]: These will only ever be updated by the server. Remove from toJSON

File: src/app/models/Tag.ts [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    };
+  }
+
+  redirectPath(): string {
```
Comment (atruskie) [inline src/app/models/Tag.ts:74]: not central to this issue, but the naming of `redirectPath` is a little weird. It implies I can only use that function will I am doing a redirect... I can't just use it navigate to the view?

File: src/app/services/baw-api/tags.service.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+function createTag(modelId: Id) {
+  return new Tag({
+    id: modelId,
+    text: "PLACEHOLDER TAG",
+    count: Math.floor(Math.random() * 10000),
+    isTaxanomic: modelId % 5 === 0,
+    typeOfTag: "general",
+    retired: modelId % 5 === 0,
+    notes: "PLACEHOLDER NOTES",
```
Comment (atruskie) [inline src/app/services/baw-api/tags.service.ts:92]: anywhere you see `notes` on one of our models, it should always returns an object/hash - it'll be a json blob in the database that gets inserted into the model's notes field

**Review by atruskie — APPROVED** [review]

File: src/app/models/Tag.ts (hunk 1 above)
Comment (Allcharles) [reply src/app/models/Tag.ts:70]: Extracted to new issue: https://github.com/QutEcoacoustics/workbench-client/issues/173

File: src/app/models/Tag.ts (hunk 2 above)
Comment (Allcharles) [reply src/app/models/Tag.ts:74]: Extracted to new issue: https://github.com/QutEcoacoustics/workbench-client/issues/173

File: src/app/services/baw-api/tags.service.ts (hunk 3 above)
Comment (Allcharles) [reply src/app/services/baw-api/tags.service.ts:92]: Extracted to new issue: https://github.com/QutEcoacoustics/workbench-client/issues/173

**Pull request merged**